vendor/modules.txt

```diff
@@ -998,6 +998,9 @@ golang.org/x/crypto/pkcs12/internal/rc2
 golang.org/x/crypto/salsa20/salsa
 golang.org/x/crypto/ssh
 golang.org/x/crypto/ssh/internal/bcrypt_pbkdf
+# golang.org/x/exp v0.0.0-20221006183845-316c7553db56 => golang.org/x/exp v0.0.0-20221006183845-316c7553db56
```

Comment:
Adding this dep everywhere is really unfortunate. What if you copy-paste the constraint instead of importing this? I think nothing bad would happen?

Reply:
I think as generics are used more and more in projects, generic constraints will also be used in many places, so the import of the package seems acceptable. It should be noted, however, that this package is experimental.

Reply:
> I think nothing bad would happen?

Not right now, but if `Ordered` gets extended upstream later on, the local copy will be stale. This won't have an impact until someone tries to use the generic function with a type that is considered ordered by golang.org/x/exp but not our copy.

Although the impact will be minimal and can be rectified at that time, I think we should bite the bullet now and add golang.org/x/exp as dependency of Kubernetes. As @weilaaa said, we may need it again elsewhere.

Reply:
> I don't think go adds types very often and if they do, it's an easy fix & backport.

I'm worried that we'll do this workaround (copy the interface) in several different places once we start to use generics more frequently. But I'm happy to defer to @liggitt  for the final decision.